### ISSUE TITLE
🔐 Login With Weird

### ISSUE THREAD

**Issue by zicklag** [commune-os/weird#25]
An important part of Weird is the ability to log into other websites with your Weird account.

Right now Rauthy, the auth server used by Weird, is experimenting with implementing the extremely work-in-progress [FedCM](https://github.com/fedidcg/FedCM) protocol, which, if successful, might allow users to log into existing sites with their own choice of identity provider, instead of only being able to sign in with Google/Apple, etc. That's a ways out, as the spec has not yet been finalized and only one browser (Chrome) has experimental support for it.

In the meantime we also want to allow people to Login with Weird using the existing OIDC standard. In order to make it as easy as possible for other sites to add "Login with Weird" buttons, we are going to support "ephemeral clients" from the [Solid OIDC](https://solidproject.org/TR/oidc#clientids-document) spec, which allows you to set the `client_id` to a URL that returns a JSON doc matching [this Rust struct](https://github.com/sebadob/rauthy/blob/a0e756bd66d70942c5f4860c07dac354748083a7/rauthy-models/src/request.rs#L201), to automatically register the client.

In other words, this lets apps add a "Login with Weird" button _without_ having to do any specific registration with our Weird server, so that people can integrate Weird login without needing to coordinate with us or create a Weird account. This saves both us and them work.

We'll have to test to make sure that's an option for all of the services that we want to be able to integrate with Weird. It's possible not all of them will support the extension, but we can evaluate that is it comes.

**Comment by erlend-sh** [comment]
Related:
- https://github.com/sebadob/rauthy/discussions/146
- https://github.com/indieweb/indieauth/issues/133

**Comment by erlend-sh** [comment]
> we are going to support "ephemeral clients" from the Solid OIDC spec, which allows you to set the client_id to a URL that returns a JSON doc matching this Rust struct, to automatically register the client.

what will this look like in the UI? Are we doing a type of [web sign-in](https://indieweb.org/Web_sign-in)?

**Comment by zicklag** [comment]
> Are we doing a type of [web sign-in](https://indieweb.org/Web_sign-in)?

No, not yet.

> what will this look like in the UI?

There actually isn't any UI right now. It means that apps can register themselves with Weird for login, without needing a UI to do it. For example, to add a "Login with Google" button to your app, you have to create a Google account and register a new OAuth application, maybe get approval, etc. With Weird, you don't have to get any approval or anything, you can just add your config JSON in some public place.

For example, for the apps that I have connected to weird right now, I have their info stored in this git repo: https://git.weird.one/zicklag/oidc-config.

Ideally the apps should do this themselves, so that nobody needs to put JSON files anywhere, and it'd be totally automatic, but not many apps know about the ephemeral client setup, so doing it manually is a workaround to get support in apps that aren't aware of the strategy we're using.

**Comment by zicklag** [comment]
Apps that we want to integrate weird, but need fixes to work:

- **Opengist:** https://github.com/thomiceli/opengist/issues/227
- **Shiori:** https://github.com/go-shiori/shiori/discussions/461
